### PR TITLE
ag-Grid v18

### PR DESCRIPTION
**Pull request by gportela85**
https://github.com/exhi/hoist-react/issues/389

### CODE REVIEW

**Comment by lbwexler** [comment]
Thanks for this Gil.

Any breaking changes to look out for?  New features? What have we tested?

Trying this now and would like to merge

**Comment by gportela85** [comment]
@lbwexler Apparently the `gridSizeChanged` event was deprecated. This happened because ag-Grid v18 uses flexbox layouts instead of JS internal layouts. I talked to the engineers there and explained we were using that event to resize the columns and also mentioned that it broke this example page: https://www.ag-grid.com/javascript-grid-responsiveness/

They said the `gridSizeChanged` event would be back on their next minor release, so I'm waiting for that to happen. It's the only breaking change.

**Comment by lbwexler** [comment]
Wow -- this sounds like a huge change, actually

So this PR is *not* ready for consideration until the minor release is
ready?  I came *this* close to just merging it.  :-)  These comments are
pretty important!

**Comment by lbwexler** [comment]
See now this is listed as 'WIP' -- thanks for that -- maybe just a bit subtle :-)

**Comment by gportela85** [comment]
@lbwexler The `gridSizeChanged` has been re-added to ag-Grid, I verified that everything is still working like it should. Maybe you want to run some additional tests.
One thing I noticed is that we get some extra warning from ag-grid now, so if we try to call sizeColumnsToFit on a hidden grid it will complain about it, so I added some extra verifications before calling that method.

File: cmp/grid/Grid.js [hunk 1]
```diff
@@ -318,7 +318,12 @@ class Grid extends Component {
     }
 
     onGridSizeChanged = (ev) => {
-        ev.api.sizeColumnsToFit();
+        const {api} = ev;
+
+        if (api.gridPanel.getWidthForSizeColsToFit()) {
```
[the remark this answers is not in the document]
Comment (amcclain) [reply cmp/grid/Grid.js]: Gil and I discussed - it's not, but I see we've added a new `isDisplayed` getter to HoistComponent in #407 - we can merge this then use / test that here.

**Pull request merged**